content/5_Plat/Centroid.mdx

```diff
@@ -235,99 +235,139 @@ private static class Centroid {
 
 ## Solution - Xenia & Tree
 
+<!-- code from https://codeforces.com/contest/342/submission/114241456 (with a couple of minor modifications) -->
+
 ```cpp
-#include <cstdio>
-#include <cstring>
+#include <iostream>
+#include <algorithm>
 #include <vector>
-
-template<typename T> bool ckmin(T& a, const T& b){return b<a?a=b,1:0;}
-
-const int MN = 1e5+10, INF = 0x3f3f3f3f;
-
-int N, M, s[MN], m[MN][2], t, b, d;
-bool r[MN], red[MN];
-std::vector<int> a[MN], v[MN];
-struct info
-{
-	public:
-		int n,b,d;
-};
-std::vector<info> g[MN];
-
-int dfs(int n, int p=0)
-{
-	s[n]=1;
-	for(auto x:a[n])
-		if(x!=p&&!r[x])
-			s[n]+=dfs(x, n);
-	return s[n];
+using namespace std;
+ 
+const int inf = 2000000011;
+#define PB push_back
+ 
+int N, Q, W[100005], P[100005], ans[100005];
+vector<int> adj[100005];
+bool col[100005];
+
+// W[] is weight(size) of subtree (excluding deleted nodes)
+// P[] is parent in centroid tree
+// ans[] is closest red node in subtree of centroid tree
+// adj[] stores original tree given in input
+// col[] stores if node has already been colored red (optional)
+ 
+bool r[100005];
+// r[] stores if a node has been deleted
+ 
+int dfs_w(int u, int p = 0) {
+	if (r[u])
+		return 0;
```

Comment:
```suggestion
	if (r[u])
		return 0;
```

This is redundant, right? Probably can delete.

Reply:
> Looks fine. Would it be better to cut out extra log factor though?

Both Galen Colin's tutorial and the other one mention using some method to find distance. I could use sparse table to do it in constant time, but I feel like binary jumping is just more intuitive from an educational standpoint, seeing as it's part of the platinum topics. In the end, it really doesn't matter as long as the solution pass the tests.

Reply:
maybe it suffices to just mention that it's possible to cut out the extra log factor? @frodakcin @bqi343

Reply:
Just do it in the same way as the original solution does? It's quite simple and there's no extra log factor.